setup.py

```diff
@@ -89,6 +89,7 @@
         "packaging >= 14.3",
         "google-cloud-storage >= 1.32.0, < 2.0.0dev",
         "google-cloud-bigquery >= 1.15.0, < 3.0.0dev",
+        "google-api-python-client >= 2.29",  # for API discovery
```

Comment:
The Python discovery client is something we'd want to avoid adding, could we remove the project number to project ID conversion for now? We can aim to alter `CLOUD_ML_PROJECT_ID` service-side or add another env var in Custom Jobs / Pipelines that explicitly contains the project ID

Reply:
>The Python discovery client is something we'd want to avoid adding

Could you share the main issues making adding this package undesirable compared to other packages? 

>could we remove the project number to project ID conversion for now?

This is possible, but some consequences would be not quite desirable. Namely, the staging bucket names are currently in the form `<project-id>-vertex-staging-<region>`. Using project numbers would result in less readable names.

>We can aim to alter CLOUD_ML_PROJECT_ID service-side or add another env var in Custom Jobs / Pipelines that explicitly contains the project ID

I agree. That solution would be the best.

Reply:
I've removed this dependency. Let's revisit this decision later.